addon/components/frost-numeric-field.js

```diff
@@ -0,0 +1,232 @@
+
+  propTypes: {
+    // options
+    align: PropTypes.string,
+    tabindex: PropTypes.number,
+    type: PropTypes.string,
+    isIncrementControlVisible: PropTypes.bool,
+    allowNegativeValues: PropTypes.bool,
+    errorMessage: PropTypes.string,
+    class: PropTypes.string,
+
+    // Setting these as part of establishing an inital value
+    autocapitalize: PropTypes.string,
+    autofocus: PropTypes.bool,
+    autocorrect: PropTypes.string,
+    form: PropTypes.string,
+    maxlength: PropTypes.number,
+    placeholder: PropTypes.string,
+    readonly: PropTypes.bool,
+    required: PropTypes.bool,
+    selectionDirection: PropTypes.string,
+    spellcheck: PropTypes.bool,
+    value: PropTypes.string.isRequired,
+    title: PropTypes.string,
+
+    _internalErrorMessage: PropTypes.string,
+    _errorMessage: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.bool
+    ]),
+    _decreaseButtonDisable: PropTypes.bool,
+    _increaseButtonDisable: PropTypes.bool
+  },
+
+  getDefaultProps () {
+    return {
+      // options
+      align: 'left',
+      autocapitalize: 'off',
+      autocorrect: 'off',
+      autofocus: false,
+      isIncrementControlVisible: false,
+      allowNegativeValues: false,
+      isHookEmbedded: false,
+      readonly: false,
+      required: false,
+      spellcheck: false,
+      tabindex: 0,
+      type: 'text'
+    }
+  },
+
+  // == Functions =============================================================
+
+  _isNumberKey (e) {
+    const value = this.get('value')
+    let decimalExist = value && value.indexOf('.') >= 0
+    let charCode = e.which || e.keyCode
+    if (this.get('allowNegativeValues')) {
+      /* check if user can enter negative symbol*/
+
+      /* get the cursor position to see if the negative sign is placed at the front of the numbers */
+      let cursorPos = this.$()[0].getElementsByTagName('input')[0].selectionStart
+      if (String.fromCharCode(charCode) === '-' && cursorPos === 0) {
+        return true
+      }
+      return this.checkIfKeyInvalid(charCode, decimalExist)
+    } else {
+      return this.checkIfKeyInvalid(charCode, decimalExist)
+    }
+  },
+
+  checkIfKeyInvalid (charCode, decimalExist) {
+    if (String.fromCharCode(charCode) === '.' && decimalExist) {
+      return false
+    } else if (String.fromCharCode(charCode) !== '.' && charCode > 31 &&
+      (String.fromCharCode(charCode) < '0' || String.fromCharCode(charCode) > '9')) {
+      return false
+    } else {
+      return true
+    }
+  },
+
+ /** calculate how many digit a value has
+  * @param {string} value - value of the input
+  * @returns {number} - number of digits
+ */
+  _calculateDigit (value) {
+    return value.slice(value.indexOf('.') + 1).length
+  },
+
+  /** format decimal
+    * @param {string} value - value of the input
+    * @param {number} numberOfDigit - number of digit
+ */
+  _setCalculatedValue (value, numberOfDigit) {
```

Comment:
I'd probably suggest renaming this to reflect that it's not a `set` now, but we can clean that up the next time we're in the code